tutorials/klusterai-api/effective-prompt-engineering.ipynb

```diff
@@ -0,0 +1,589 @@
+      "\n",
+      "1. **Product B has the highest sales**: With 20 sales, Product B is the best-selling product in the dataset.\n",
+      "2. **Product C has the lowest sales**: With only 5 sales, Product C is the worst-selling product in the dataset.\n",
+      "3. **Product A and D have moderate sales**: Products A and D have sales of 10 and 15, respectively, which are lower than Product B but higher than Product C.\n",
+      "4. **Total sales**: If we add up the sales of all products, we get a total of 10 + 20 + 5 + 15 = 50 sales.\n",
+      "\n",
+      "Overall, the data suggests that Product B is the most popular, while Product C is the least popular. However, without more context or information about the products, market, or time period, it's difficult to draw more specific conclusions.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Define the bad prompt\n",
+    "bad_prompt_text = \"\"\"Here's some data:\n",
+    "\n",
+    "| Product | Sales |\n",
+    "|---------|-------|\n",
+    "| A       | 10    |\n",
+    "| B       | 20    |\n",
+    "| C       | 5     |\n",
+    "| D       | 15    |\n",
+    "\n",
+    "Tell me about it.\"\"\"\n",
+    "\n",
+    "# Display the prompt for visual reference\n",
+    "display(Markdown(bad_prompt_text))\n",
+    "\n",
+    "# Use the variable in the API call\n",
+    "response_bad = client.chat.completions.create(\n",
+    "    model=model,\n",
+    "    messages=[{\"role\": \"user\", \"content\": bad_prompt_text}],\n",
+    "    temperature=0.5\n",
+    ").choices[0].message.content\n",
+    "print(response_bad)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b8c92069",
+   "metadata": {},
+   "source": [
+    "### Analysis\n",
+    "The Bad prompt offers only the most rudimentary information—products and their sales—and provides no clear instructions beyond \"tell me about it.\" This inevitably results in:\n",
+    "1. **Minimal Context**: Without pricing, profit margins, or a specific goal (e.g., analyzing profitability, recommending strategies), the LLM can only produce surface-level observations.\n",
+    "2. **No Actionable Guidance**: The prompt doesn't request particular insights or next steps, so the model responds with high-level statements rather than solutions.\n",
+    "3. **Limited Utility**: Because the data is purely quantitative (sales alone) and lacks details such as time frame, product characteristics, or marketing context, the response cannot go beyond pointing out which products sold more or less.\n",
+    "\n",
+    "Overall, this prompt's lack of clear direction and supporting information stifles the model's ability to provide meaningful analysis or recommendations."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "87dab731",
+   "metadata": {},
+   "source": [
+    "## 🔵 Better Prompt\n",
+    "\n",
+    "The below better prompt provides more context and clearer instructions, however, there is still room for improvement:"
```

Comment:
```suggestion
    "The prompt below provides better context and clearer instructions, however, there is still room for improvement:"
```

Reply:
Add a bit more details on why it is better. Don't mention that there is room for improvement here, but after you present the resutls to the user you should state "Why the result was not good enough, and that there is room for improvement"